### ISSUE TITLE
JAX-RS and JAXB 3.0 serialization issue

### ISSUE THREAD

**Issue by rlfnb** [eclipse-ee4j/jersey#4553]
Attached a PoC to reproduce the 
`
[2020-08-21T18:25:12.249+0200] [Payara 5.2020.4] [SEVERE] [] [org.glassfish.jersey.message.internal.WriterInterceptorExecutor] [tid: _ThreadID=103 _ThreadName=http-thread-pool::http-listener-1(2)] [timeMillis: 1598027112249] [levelValue: 1000] [[
  MessageBodyWriter not found for media type=application/xml, type=class com.example.MyClass, genericType=class com.example.MyClass.]]
`
Basically, I tried to do it as described [here](https://javaee.github.io/tutorial/jaxrs-advanced007.html). 
Used Maven 3.6.1 and Java11.

[JAXRSJAXBPoC.zip](https://github.com/eclipse-ee4j/jersey/files/5109964/JAXRSJAXBPoC.zip)

**Comment by jansupol** [comment]
Are by any chance combining Jersey 2.x and JAX-B 3.0? That will never work. You would need Jersey 3.x with JAX-B 3.x.

**Comment by rlfnb** [comment]
@jansupol ACK, maybe good to have it as a starting point for jaxb3 only?

**Comment by jansupol** [comment]
Sorry, I do not understand what you mean. Jersey 3.0.0.-M6 is available in maven central, for testing purposes.
